fred2/fredview.cpp

```diff
@@ -966,23 +968,31 @@ void CFREDView::OnLButtonDown(UINT nFlags, CPoint point)
 	drag_rotate_save_backup();
 	
 	if (nFlags & MK_CONTROL)	{  // add a new object
-		if (!Bg_bitmap_dialog) {
-			if (on_object == -1) {
-				Selection_lock = 0;  // force off selection lock
-				on_object = create_object_on_grid(waypoint_instance);
-
-			} else
-				Dup_drag = 1;
+		bool shift_down = (GetAsyncKeyState(VK_SHIFT) & 0x8000) != 0;
 
+		if (shift_down) {
+			Selection_lock = 0; // force off selection lock
+			on_object = create_object_on_grid(waypoint_instance, true);
 		} else {
```

Comment:
This is a handling change for FRED, isn't it?
Is that intentional, and if so, why, or am I just misreading things?